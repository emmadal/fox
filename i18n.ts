import { getLocales } from "expo-localization";
import { I18n } from "i18n-js";
import * as en from "./locales/en.json";
import * as fr from "./locales/fr.json";

// Set the key-value pairs for the different languages you want to support.
const translations = { en, fr };
const i18n = new I18n(translations);

const CheckPhoneLanguage = () =>
  getLocales()?.[0]?.languageCode?.match(/^(en|fr)$/i)
    ? getLocales()[0].languageCode || "en"
    : "en";

// Set the locale once at the beginning of your app.
i18n.locale = CheckPhoneLanguage();
// When a value is missing from a language it'll fall back to another language with the key present.
i18n.enableFallback = true;

export default i18n;
