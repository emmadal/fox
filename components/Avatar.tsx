import React from "react";
import { View, Text } from "react-native";
import { Image, useImage } from "expo-image";
import i18n from "@/i18n";
export type AvatarProps = {
  source: string;
  size?: number;
  className?: string;
  position?: "center" | "flex-end" | "flex-start" | "baseline" | "auto";
};

const Avatar = ({
  source,
  size,
  className,
  position = "auto",
}: AvatarProps) => {
  const image = useImage(source, {
    maxWidth: size,
    maxHeight: size,
  });
  if (!image) {
    return <Text className="dark:white text-base">{i18n.t("loading")}</Text>;
  }
  return (
    <View
      className={`overflow-hidden rounded-full border-4 border-gray-100 justify-center items-center ${className}`}
      style={[{ width: size, height: size, alignSelf: position }]}
      testID="avatar-view"
    >
      <Image
        transition={2000}
        source={{ uri: source }}
        style={{ height: size, width: size }}
        cachePolicy="memory"
        className="rounded-full"
        testID="avatar-image"
      />
    </View>
  );
};

export default Avatar;
