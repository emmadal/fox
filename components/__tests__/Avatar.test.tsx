import React from "react";
import { render } from "@testing-library/react-native";
import Avatar from "../Avatar"; // Adjust the path to your component
import i18n from "@/i18n";

// Mock expo-image globally
jest.mock("expo-image", () => ({
  ...jest.requireActual("expo-image"),
  useImage: jest.fn(),
}));

describe("Avatar Component", () => {
  beforeEach(() => {
    const { useImage } = require("expo-image");
    useImage.mockReset(); // Reset mocks between tests
  });

  it("renders correctly (snapshot)", () => {
    const { toJSON } = render(<Avatar source="https://unavatar.io/johndoe" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders loading text when image is not ready", () => {
    const { useImage } = require("expo-image");
    useImage.mockReturnValue(null);
    const { getByText } = render(
      <Avatar source="https://unavatar.io/johndoe" />,
    );
    expect(getByText(i18n.t("loading"))).toBeTruthy();
  });

  it("renders image correctly when ready", () => {
    const { useImage } = require("expo-image");
    useImage.mockReturnValue(true); // Simulate image being loaded

    const { getByTestId } = render(
      <Avatar
        source="https://unavatar.io/johndoe"
        size={100}
        className="custom-class"
        position="center"
      />,
    );

    const avatarView = getByTestId("avatar-view");
    const avatarImage = getByTestId("avatar-image");

    expect(avatarView.props.style).toEqual(
      expect.arrayContaining([
        { width: 100, height: 100, alignSelf: "center" },
      ]),
    );

    // Access the first element in the array
    expect(avatarImage.props.source[0]).toEqual({
      uri: "https://unavatar.io/johndoe",
    });

    expect(avatarImage.props.style).toMatchObject({ width: 100, height: 100 });
  });

  it("applies the correct default position when no position is provided", () => {
    const { useImage } = require("expo-image");
    useImage.mockReturnValue(true);

    const { getByTestId } = render(
      <Avatar source="https://unavatar.io/johndoe" size={50} />,
    );
    const avatarView = getByTestId("avatar-view");

    expect(avatarView.props.style).toEqual(
      expect.arrayContaining([{ alignSelf: "auto", width: 50, height: 50 }]),
    );
  });
});
