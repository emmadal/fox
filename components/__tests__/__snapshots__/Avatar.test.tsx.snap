// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders correctly (snapshot) 1`] = `
<Text
  className="dark:white text-base"
>
  Loading
</Text>
`;
